agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java

```diff
@@ -270,6 +274,92 @@ private static void overlayConfiguration(
+        for (Configuration.ProcessorAttribute attribute : processor.include.attributes) {
+          attribute.key = mapAttributeKey(attribute.key);
+        }
+      }
+      if (processor.exclude != null && processor.type == Configuration.ProcessorType.ATTRIBUTE) {
+        for (Configuration.ProcessorAttribute attribute : processor.exclude.attributes) {
+          attribute.key = mapAttributeKey(attribute.key);
+        }
+      }
+      for (Configuration.ProcessorAction action : processor.actions) {
+        if (action.key != null && processor.type == Configuration.ProcessorType.ATTRIBUTE) {
+          action.key = AttributeKey.stringKey(mapAttributeKey(action.key.getKey()));
+        }
+      }
+      if (processor.name != null && processor.name.fromAttributes != null) {
+        List<String> newFromAttributes = new ArrayList<>();
+        for (String oldFromAttribute : processor.name.fromAttributes) {
+          String newFromAttribute = mapAttributeKey(oldFromAttribute);
+          newFromAttributes.add(newFromAttribute);
+        }
+        processor.name.fromAttributes = newFromAttributes;
+      }
+      if (processor.body != null && processor.body.fromAttributes != null) {
+        List<String> newFromAttributes = new ArrayList<>();
+        for (String oldFromAttribute : processor.body.fromAttributes) {
+          String newFromAttribute = mapAttributeKey(oldFromAttribute);
+          newFromAttributes.add(newFromAttribute);
+        }
+        processor.body.fromAttributes = newFromAttributes;
+      }
+    }
+  }
+
+  private static String mapAttributeKey(String oldAttributeKey) {
+    // Common attributes across HTTP client and server spans
+    if (oldAttributeKey.equals(SemanticAttributes.HTTP_METHOD.getKey())) {
+      return SemanticAttributes.HTTP_REQUEST_METHOD.getKey();
+    } else if (oldAttributeKey.equals(SemanticAttributes.HTTP_STATUS_CODE.getKey())) {
+      return SemanticAttributes.HTTP_RESPONSE_STATUS_CODE.getKey();
+    } else if (oldAttributeKey.startsWith("http.request.header.")
+        || oldAttributeKey.startsWith("http.response.header.")) {
+      return oldAttributeKey.replace('_', '-');
+    } else if (oldAttributeKey.equals(SemanticAttributes.NET_PROTOCOL_NAME.getKey())) {
+      return SemanticAttributes.NETWORK_PROTOCOL_NAME.getKey();
+    } else if (oldAttributeKey.equals(SemanticAttributes.NET_PROTOCOL_VERSION.getKey())) {
+      return SemanticAttributes.NETWORK_PROTOCOL_VERSION.getKey();
+    } else if (oldAttributeKey.equals(SemanticAttributes.NET_SOCK_PEER_ADDR.getKey())) {
+      return SemanticAttributes.NETWORK_PEER_ADDRESS.getKey();
+    } else if (oldAttributeKey.equals(SemanticAttributes.NET_SOCK_PEER_PORT.getKey())) {
+      return SemanticAttributes.NETWORK_PEER_PORT.getKey();
+    }
+
+    // HTTP client span attributes
+    // http.url is handled via LazyHttpUrl
+    if (oldAttributeKey.equals(SemanticAttributes.HTTP_RESEND_COUNT.getKey())) {
+      return "http.request.resend_count";
```

Comment:
is there no constant for this?

Reply:
I see constant being used in upstream repo: https://github.com/search?q=repo%3Aopen-telemetry%2Fopentelemetry-java-instrumentation%20http_request_resend_count&type=code

Reply:
we used the copy from the standalone exporter. this constant is not available.

Reply:
// this is a copy of io.opentelemetry.semconv.SemanticAttributes
// because the module that contains that class is not stable,
// so don't want to take a dependency on it

when can we use the upstream copy?

Reply:
we only need to use the copy in the azure sdk repo

Reply:
can you be more specific about it? SemanticAttributes in https://mvnrepository.com/artifact/io.opentelemetry/opentelemetry-semconv/1.30.1-alpha is deprecated.

Reply:
import io.opentelemetry.semconv.SemanticAttributes; 1.21 missing some other attributes. 1.30 is deprecated.